overlord/state/state_test.go

```diff
@@ -1039,14 +1039,15 @@ func (ss *stateSuite) TestPruneHonorsStartOperationTime(c *C) {
 	c.Check(chg.Status(), Equals, state.HoldStatus)
 }
 
-func (ss *stateSuite) TestReadStateInitsCache(c *C) {
+func (ss *stateSuite) TestReadStateInitsTransientMapFields(c *C) {
 	st, err := state.ReadState(nil, bytes.NewBufferString("{}"))
 	c.Assert(err, IsNil)
 	st.Lock()
 	defer st.Unlock()
 
 	st.Cache("key", "value")
 	c.Assert(st.Cached("key"), Equals, "value")
+	st.RegisterPendingChangeByAttr("attr", func(*state.Change) bool { return false })
```

Comment:
It took me a moment to realize why this got added, state.ReadState()  and state.New() initialize state.State differently for me a comment here would have been helpful but probably overkill because TestReadStateInitsTransientMapFields gives hints here.